spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -168,6 +176,100 @@ def send_login_request(user, password)
       end
     end
 
+    def handle_two_step(response)
+      @x_apple_web_session_token = response["x-apple-web-session-token"]
+      @scnt = response["scnt"]
+
+      r = request(:get) do |req|
+        req.url "https://idmsa.apple.com/appleauth/auth"
+        req.headers["scnt"] = @scnt
+        req.headers["X-Apple-Web-Session-Token"] = @x_apple_web_session_token
+        req.headers["Accept"] = "application/json"
+      end
+
+      if r.body.kind_of?(Hash) && r.body["trustedDevices"].kind_of?(Array)
+        if r.body.fetch("securityCode", {})["tooManyCodesLock"].to_s.length > 0
+          raise ITunesConnectError.new, "Too many verification codes have been sent. Enter the last code you received, use one of your devices, or try again later."
+        end
+
+        old_client = (begin
+                        Tunes::RecoveryDevice.client
+                      rescue
+                        nil # since client might be nil, which raises an exception
+                      end)
+        Tunes::RecoveryDevice.client = self # temporary set it as it's required by the factory method
+        devices = r.body["trustedDevices"].collect do |current|
+          Tunes::RecoveryDevice.factory(current)
+        end
+        Tunes::RecoveryDevice.client = old_client
```

Comment:
This can be re-written as:
```ruby
devices = r.body["trustedDevices"].map do |current|
  device = Tunes::RecoveryDevice.new(current)
  device.client = self
end
```
no need to get `old_client` or set the class-level `client`